src/main/webapp/app/exercises/programming/hestia/git-diff-report/git-diff-report.component.html

```diff
@@ -52,16 +52,25 @@ <h4 class="ps-2">
         />
     </div>
 </div>
-@for (filePath of filePaths; track filePath) {
-    <div class="mt-2">
-        @if (entriesByPath.get(filePath)?.length) {
-            <jhi-git-diff-file-panel
-                [diffForTemplateAndSolution]="diffForTemplateAndSolution"
-                [diffEntries]="entriesByPath.get(filePath) ?? []"
-                [templateFileContent]="templateFileContentByPath.get(filePath)"
-                [solutionFileContent]="solutionFileContentByPath.get(filePath)"
-                [numberOfContextLines]="numberOfContextLines"
-            />
-        }
+<div [hidden]="!allDiffsReady">
```

Comment:
Why do you use `<div [hidden]="!allDiffsReady">` instead of `@if(...)` ?

Reply:
The difference between the two is that @if would completely remove the panels (which compute the diff via the editor) from the document, preventing the diff from being computed.

If I just hide them, the computation still occurs (and eventually unhides the finished panels)

Reply:
Makes sense. Thanks for clarifying!